standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -4109,7 +4110,7 @@ private List<Partition> add_partitions_core(final RawStore ms, String catName,
             + TableName.getQualified(catName, dbName, tblName) +
             " does not exist");
       }
-
+      MTable mTable = getMS().ensureGetMTable(catName, dbName, tblName);
```

Comment:
should we do an evaluation even if later conditions won't trigger updatePartitonColStatsInternal?